storage/src/main/scala/coop/rchain/storage/Lmdb.scala

```diff
@@ -0,0 +1,1282 @@
+/*               __          __                                         *\
+**    __________/ /_  ____ _/_/___                                      **
+**   / ___/ ___/ __ \/ __ `/ / __ \     RChain API                      **
+**  / /  / /__/ / / / /_/ / / / / /     (c) http://rchain.coop          **
+** /_/   \___/_/ /_/\____/_/_/ /_/                                      **
+\*                                                                      */
+
+package coop.rchain.Storage
+
+import java.io.File
+import java.nio.ByteBuffer
+import org.lmdbjava.DbiFlags.{MDB_CREATE, MDB_DUPSORT}
+import org.lmdbjava.Env.create
+import org.lmdbjava.EnvFlags._
+import org.lmdbjava.GetOp._
+import org.lmdbjava.PutFlags.MDB_NODUPDATA
+import org.lmdbjava.SeekOp._
+import org.lmdbjava._
+import scala.collection.mutable.ArrayBuffer
+
+
+class Lmdb(_dirNameIn:Option[String], _nameIn:Option[String],
+           _isKeyToValuesIn:Boolean = false,
+           _isWritableIn:Boolean = false,
+           _baseDirIn:Option[String] = None,
+           _maxSizeIn:Long = Lmdb.minDbSize,
+           _maxStrValueLength:Int = 0) {
+
+  assert(_dirNameIn.isDefined && !_dirNameIn.get.isEmpty)
+  assert(_nameIn.isDefined && !_nameIn.get.isEmpty)
+  assert(!_baseDirIn.isDefined || !_baseDirIn.get.isEmpty)
+  assert(Lmdb.minDbSize <= _maxSizeIn && _maxSizeIn <= Lmdb.maxDbSize)
+
+  protected[Storage] val dirNameIn =
+    if (_dirNameIn.isDefined && !_dirNameIn.get.isEmpty) _dirNameIn.get
+    else { throw new RChainException("DB Directory is invalid") }
+  protected[Storage] val nameIn =
+    if (_nameIn.isDefined && !_nameIn.get.isEmpty) { _nameIn.get }
+    else { throw new RChainException("Name is invalid") }
+  protected[Storage] val isKeyToValuesIn = _isKeyToValuesIn
+  protected[Storage] val isWritableIn = _isWritableIn
+  protected[Storage] val baseDirIn =
+    if (_baseDirIn.isDefined) {
+      if (new File(_baseDirIn.get).isDirectory) { _baseDirIn.get }
+      else { throw new RChainException("Base Directory is invalid")}
+    }
+    else System.getProperty("user.dir") // not our problem if throws
+  protected[Storage] val mapSizeIn: Long =
+    if (Lmdb.maxDbSize < _maxSizeIn) {
+      Lmdb.maxDbSize
+    }
+    else if (_maxSizeIn < Lmdb.minDbSize) {
+      Lmdb.minDbSize
+    }
+    else {
+      _maxSizeIn
+    }
+
+  protected[Storage] val dbDir = new File(baseDirIn + "/" + dirNameIn)
+  private var success = false
+  if (!dbDir.exists) {
+    success = dbDir.mkdir()
```

Comment:
`mkdirs()` maybe, unless you want to limit it to one away.

Reply:
There is only one directory needed.  Is there a reason to use mkdirs()?

Reply:
if `dirNameIn` is something like `tests/one`, then `mkdirs()` will also make the parent when necessary. It's okay with me if you want to disallow that.